mayastor/src/lvs/lvol.rs

```diff
@@ -292,7 +292,6 @@ impl Lvol {
 
         // we must always unshare before destroying bdev
         let _ = self.unshare().await;
-        self.wipe_super().await?;
```

Comment:
More a n00b question rather than a comment on your changes: how do we ensure no user data gets leaked? I see that `wipe_super` zeroes the first 4MB of the data. Why only 4MB? :thinking:

Reply:
`wipe_super` is also zeroing out the superblock such that it would make any filesystem already on the disk unusable.

I'd have to defer to @gila as to why we chose zero out 4MB of the data partition, but in general we didn't want to be zeroing out the entire disk as this would be time consuming. Specifically from the commit message (https://github.com/openebs/Mayastor/commit/40545fff42055db242ad1bc10463b3ebf6951b59):
"When the base bdev does not support UNMAP, writing zero's is costly and very time consuming"

Also, the creation/deletion can be called from a gRPC request, so if the operations are taking a significant amount of time the gRPC calls can timeout. This change is actually trying to address this as we were seeing timeouts in the control plane due to this.

Reply:
If we don't do this on destroy, should we reduce the DestroyVolume specific timeout?

Reply:
Yes I think we should do that. Then, if at a later stage we start seeing the DestroyVolume timing out, we'll have a relatively quick feedback loop.

Reply:
Thanks @paulyoong for explaining in detail. I've a good understanding of your changes now and why we avoid zeroing out the whole volume (which I kinda guessed already) but my main question remains: how do we ensure no user data gets leaked? Or is that not a concern?

Reply:
by "leaked", I mean remaining on the drive, in case no future volumes overwrite it, which could be sensitive. Perhaps data is always encrypted?

Reply:
You are correct - at the moment the data remains on the disk. I remember there being a discussion about this previously and I think it was decided that this should be addressed in the future.